src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -2222,5 +2236,16 @@ internal static void UpdateTypeChecking(this FileContentManager file, Compilatio
                 compilation.ExitWriteLock();
             }
         }
+
+        private static void CheckForGlobalCycleChange(FileContentManager file, List<Diagnostic> diagnostics)
+        {
+            var numRemovedCycleDiags = file.CurrentSemanticDiagnostics().Count(m => m.Code == ErrorCode.InvalidCyclicTypeParameterResolution.Code())
+                - diagnostics.Count(m => m.Code == ErrorCode.InvalidCyclicTypeParameterResolution.Code());
+
+            if (numRemovedCycleDiags != 0 || diagnostics.Any(x => x.Source != file.FileName.Value))
```

Comment:
If `numRemovedCycleDiags` can be negative, I think it'd be better to call it something like `numCycleDiagnosticsChange`.

Reply:
done